frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts

```diff
@@ -44,22 +44,27 @@ import {
-import {WorkPackageViewTimelineService} from "core-app/modules/work_packages/routing/wp-view-base/view-services/wp-view-timeline.service";
-import {WorkPackageRelationsService} from "core-components/wp-relations/wp-relations.service";
-import {WorkPackageViewHierarchiesService} from "core-app/modules/work_packages/routing/wp-view-base/view-services/wp-view-hierarchy.service";
-import {WorkPackageTimelineCell} from "core-components/wp-table/timeline/cells/wp-timeline-cell";
-import {selectorTimelineSide} from "core-components/wp-table/wp-table-scroll-sync";
-import {debugLog, timeOutput} from "core-app/helpers/debug_output";
-import {RenderedWorkPackage} from "core-app/modules/work_packages/render-info/rendered-work-package.type";
-import {HalEventsService} from "core-app/modules/hal/services/hal-events.service";
-import {WorkPackageNotificationService} from "core-app/modules/work_packages/notifications/work-package-notification.service";
-import {combineLatest} from "rxjs";
-import {UntilDestroyedMixin} from "core-app/helpers/angular/until-destroyed.mixin";
-import {WorkPackagesTableComponent} from "core-components/wp-table/wp-table.component";
+import {input, InputState} from 'reactivestates';
+import {WorkPackageTable} from 'core-components/wp-fast-table/wp-fast-table';
+import {WorkPackageTimelineCellsRenderer} from 'core-components/wp-table/timeline/cells/wp-timeline-cells-renderer';
+import {States} from 'core-components/states.service';
+import {WorkPackageViewTimelineService} from 'core-app/modules/work_packages/routing/wp-view-base/view-services/wp-view-timeline.service';
+import {WorkPackageRelationsService} from 'core-components/wp-relations/wp-relations.service';
+import {WorkPackageViewHierarchiesService} from 'core-app/modules/work_packages/routing/wp-view-base/view-services/wp-view-hierarchy.service';
+import {WorkPackageTimelineCell} from 'core-components/wp-table/timeline/cells/wp-timeline-cell';
+import {selectorTimelineSide} from 'core-components/wp-table/wp-table-scroll-sync';
+import {debugLog, timeOutput} from 'core-app/helpers/debug_output';
+import {RenderedWorkPackage} from 'core-app/modules/work_packages/render-info/rendered-work-package.type';
+import {HalEventsService} from 'core-app/modules/hal/services/hal-events.service';
+import {WorkPackageNotificationService} from 'core-app/modules/work_packages/notifications/work-package-notification.service';
+import {combineLatest, Observable} from 'rxjs';
+import {UntilDestroyedMixin} from 'core-app/helpers/angular/until-destroyed.mixin';
+import {WorkPackagesTableComponent} from 'core-components/wp-table/wp-table.component';
+import {
+  groupIdFromIdentifier,
+  groupTypeFromIdentifier
+} from 'core-components/wp-fast-table/builders/modes/grouped/grouped-rows-helpers';
+import {WorkPackageViewCollapsedGroupsService} from 'core-app/modules/work_packages/routing/wp-view-base/view-services/wp-view-collapsed-groups.service';
 
```

Comment:
<table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/opf/openproject/blob/eb626ce32472e26aea067b18a858c6824b1aed48/frontend/src/app/components/wp-table/timeline/container/wp-timeline-container.directive.ts#L68"></span>Please use <code>ChangeDetectionStrategy.OnPush</code> for this component</td>
    </tr>
  </tbody>
</table>